3.assess_generalizability/nbconverted/ks_test_cell_lines.py

```diff
@@ -96,20 +102,71 @@
 # In[5]:
 
 
+# Split data by institution and Null genotype for comparison
+institution_Null_1_norm = plate_6_norm[
+    (plate_6_norm["Metadata_Institution"] == "iNFixion") & 
+    (plate_6_norm["Metadata_genotype"] == "Null")
+]
+institution_Null_2_norm = plate_6_norm[
+    (plate_6_norm["Metadata_Institution"] == "MGH") & 
+    (plate_6_norm["Metadata_genotype"] == "Null")
+]
+
+# Perform KS-test for each feature for the Null genotype
+Null_ks_test_results_norm = {}
+
+for column in plate_6_norm.columns:
+    if column.startswith("Metadata_"):
+        continue
+    ks_stat, p_value = stats.kstest(
+        institution_Null_1_norm[column], institution_Null_2_norm[column]
+    )
+    Null_ks_test_results_norm[column] = {"ks_stat": ks_stat, "p_value": p_value}
+
+# Convert results to DataFrame for better visualization
+Null_ks_test_results_norm_df = (
+    pd.DataFrame(Null_ks_test_results_norm)
+    .T.reset_index()
+    .rename(columns={"index": "feature"})
+)
+
+
+# In[6]:
+
+
+# Add genotype column to each KS-test results DataFrame
+WT_ks_test_results_norm_df["genotype_comparison"] = "WT"
+Null_ks_test_results_norm_df["genotype_comparison"] = "Null"
+
+# Combine the two DataFrames
+ks_test_results_norm_df = pd.concat([WT_ks_test_results_norm_df, Null_ks_test_results_norm_df], ignore_index=True)
```

Comment:
It's not obvious to me based on these changes, but do you retain the original comparison? (all mgh vs all infixion)?

Reply:
I do not, but I will go back and retain it, and then redo figure 6 to go back how it was.